src/state/profile/index.tsx

```diff
@@ -56,28 +59,120 @@ export const profileSlice = createSlice({
     },
   },
   extraReducers: (builder) => {
+    builder.addCase(fetchProfileUsername.pending, (state, action) => {
+      const { account } = action.meta.arg
+      if (state.profileAvatars[account]) {
+        state.profileAvatars[account] = {
+          ...state.profileAvatars[account],
+          usernameFetchStatus: ProfileAvatarFetchStatus.FETCHING,
+        }
+      } else {
+        state.profileAvatars[account] = {
+          hasRegistered: false,
+          username: null,
+          nft: null,
+          // I think in theory this else should never be reached since we only check for username after we checked for profile/avatar
+          // just in case I set isFetchingAvatar will be ProfileAvatarFetchStatus.FETCHED at this point to avoid refetching
+          usernameFetchStatus: ProfileAvatarFetchStatus.FETCHING,
+          avatarFetchStatus: ProfileAvatarFetchStatus.FETCHED,
```

Comment:
How does this username thunk being `pending` relate to `avatarFetchStatus` being `FETCHED`?

This point goes throughout these thunks - it seems we're consistently overwirting the loading state for the other fetch status within these `else` conditions.